drivers/include/periph/wdt.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *           (C) 2019 Inria
+ *           (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_wdt WDT
+ * @ingroup     drivers_periph
+ * @brief       watchdog timer peripheral driver
+ *
+ * @{
+ *
+ * @file        wdt.h
+ * @brief       watchdog peripheral interface definitions
+ *
+ * @author      Thomas Geithner <thomas.geithner@dai-labor.de>
+ *              Francisco Molina <francois-xavier.molina@inria.fr>
+ *              Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef PERIPH_WDT_H
+#define PERIPH_WDT_H
+
+#include <stdint.h>
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Signature for the watchdog early warning callback
+ *
+ * @param[in] arg           optional argument which is passed to the
+ *                          callback
+ */
+typedef void(*wdt_cb_t)(void *arg);
+
+/**
+ * @brief    Start watchdog timer
+ */
+void wdt_start(void);
+
+/**
+ * @brief    Stop watchdog timer
+ */
+void wdt_stop(void);
+
+/**
+ * @brief    Reset the watchdog timer counter, delay system reset
+ */
+void wdt_kick(void);
```

Comment:
I think that analogy is needlessly cruel. Instead of perpetually beating the watchdog unconscious I think we should rather `wdt_feed` it and have it reset the system when it starves.

We could also `wdt_pet` it, but I think the behavior when stopped being petted is not universal among all watchdogs.

Reply:
I wasn't actually trying to make an analogy when naming it 'wdg_kick', I was thinking of kick as delay/postpone. But I get your point, I don't mind changing it to feed (although starvation doesn't seem to me like a nice outcome either hahaha)

Reply:
Hmm, "kick" seems to be the terminus used on the wikipedia [page](https://en.wikipedia.org/wiki/Watchdog_timer). It is mentioned as an alternative to "restart".

Reply:
In other references.

> Kicking the dog
> 
> 
> The process of restarting the watchdog timer's counter is sometimes called "kicking the dog." The appropriate visual metaphor is that of a man being attacked by a vicious dog. If he keeps kicking the dog, it can't ever bite him. But he must keep kicking the dog at regular intervals to avoid a bite. Similarly, the software must restart the watchdog timer at a regular rate, or risk being restarted.

It does seem to be the defacto word used in must watchdog implementations so I would keep it.